corehq/motech/repeaters/models.py

```diff
@@ -215,6 +215,32 @@ class SQLRepeater(models.Model):
     is_paused = models.BooleanField(default=False)
     next_attempt_at = models.DateTimeField(null=True, blank=True)
     last_attempt_at = models.DateTimeField(null=True, blank=True)
+    repeater_type = models.CharField(max_length=64, db_index=True)
+
+    # attributes used in CaseRepeaters and it's subclasses
```

Comment:
I think we had discussed using a `JSONField` to store data that varies by repeater type. Why did you decide to add fields for specific types of repeaters here instead?

Reply:
The rationale behind adding separate fields instead of JSON fields was that the additional properties were less in number and it would maintain consistency with the existing code that is already written. 
Do you think that having a `JSONField` would be much beneficial here? I understand that it would give us the flexibility to add new properties in the future but it would mean updating every place where these properties are referenced.

Reply:
I think it would be a better design to only have fields on each repeater that make sense for that type of repeater.

> it would mean updating every place where these properties are referenced.

I don't think so. We can do it in a way that allows properties to be defined on `SQLRepeater` subclasses. For example:

```py
class SQLRepeater(models.Model):
    ...
    options = JSONField(default=dict)


class OptionValue(property):
    NOT_SET = object()

    def __init__(self, default=NOT_SET, choices=None):
        self.default = default
        self.choices = choices

    def __set_name__(self, owner, name):
        self.name = name

    def __get__(self, obj, objtype=None):
        if obj is None:
            return self
        if self.name in obj.options:
            return obj.options[self.name]
        if self.default is self.NOT_SET:
            raise AttributeError(self.name)
        value = self.default() if callable(self.default) else self.default
        obj.options[self.name] = value
        return value

    def __set__(self, obj, value):
        if self.choices and value not in self.choices:
            raise ValueError(f"{value!r} not in {self.choices!r}")
        self.obj.options[self.name] = value


REPEATER_FORMAT_OPTIONS = ['case_json', 'case_xml']


class SQLCaseRepeater(SQLRepeater):
    ...
    # NOTE `choices` are slightly simplified. They do not have labels,
    # which I think are only used by forms. Not sure if we need that here.
    version = OptionValue(choices=LEGAL_VERSIONS, default=V2)
    white_listed_case_types = OptionValue(default=list)
    black_listed_users = OptionValue(default=list)
    format = OptionValue(choices=REPEATER_FORMAT_OPTIONS)


class SQLFormRepeater(SQLRepeater):
    ...
    include_app_id_param = OptionValue(default=True)
    white_listed_case_types = OptionValue(default=list)
    format = OptionValue(choices=REPEATER_FORMAT_OPTIONS)
```